cli/daemon/run/testdata/echo_client/client.ts

```diff
@@ -218,8 +221,8 @@ export namespace echo {
 
             //Populate the return object from the JSON body and received headers
             const rtn = await resp.json() as HeadersData
-            rtn.Int = parseInt(resp.headers.get("x-int"), 10)
-            rtn.String = resp.headers.get("x-string")
+            rtn.Int = parseInt((resp.headers.get("x-int") ?? ""), 10)
```

Comment:
I'm not sure if we should default to `0` over `NaN`.... `NaN`feels like the right error to occur - because this field _should always be set_ in the response from the server (i.e. we don't have an `omitempty` option for headers)

So if you get NaN something has gone seriously wrong.

Reply:
`NaN` is incredibly error prone and will lead to seriously hard-to-debug problems. Keep in mind that server and client are not always 100% in sync. 

Basically any value is better than `NaN`. We can decide if that’s `0` or `undefined`. I guess `undefined` better matches the default JavaScript  behavior of “we expected a value here but it’s missing”

Reply:
So there are two things in my head here;

1. If we go with undefined - then it means we need to update the Typescript to say that field may be undefined - which is not the intended behaviour for that data structure at all. If we don't go with undefined.

2. If we return a value, then NaN will cause any calculations the number is used in to also become NaN. In my experience this is a fair better fail-safe position to be in, then to default to zero and potentially create results from those calculations which are actually incorrect - but don't appear to be.

I think this solution here is actually to throw an error if the header is `Null`; as the server did not respond as the client application requires and expects. This also aligns with the principle of least surprise for the end users.

Reply:
I don’t think that behavior is consistent with how JavaScript and TypeScript is used in practice. TypeScript is just expressing what the types should be. JS/TS are already used to getting `undefined` in places where they expected a value but the server they called didn’t match their expectations. Setting the corresponding value to `undefined` is the right thing to do. At most we should report an error in the console or whatever (assuming `console.error` exists in their runtime).

Go is the same, in a way: it will happily accept JSON and HTTP requests with missing fields and headers and report something reasonable.

Reply:
And we should not change the type annotation. Something can always go wrong, the server could return XML for all we know, but that doesn’t mean we should make all types `any`.

Reply:
Let's park this and chat about it in our 1:1

Reply:
Given that we want to get it out on Friday we should decide sooner. Happy to chat about it today or tomorrow.